coderedcms/views.py

```diff
@@ -125,3 +135,120 @@ def robots(request):
         {'robots': robots},
         content_type='text/plain'
     )
+
+@transaction.atomic()
+def import_pages(import_data, parent_page):
+    """
+    Overwrite of the wagtailimportexport `import_page` function to handle generic csvs.
+    The standard `import_pages` assumes that your pages will have a pk from the exported
+    json files.  It does not facilitate the idea that the pages you import will be
+    new pages.
+    """
+
+    pages_by_original_id = {}
+
+    # First create the base Page records; these contain no foreign keys, so this allows us to
+    # build a complete mapping from old IDs to new IDs before we go on to importing the
+    # specific page models, which may require us to rewrite page IDs within foreign keys / rich
+    # text / streamfields.
+    page_content_type = ContentType.objects.get_for_model(Page)
+
+    for (i, page_record) in enumerate(import_data['pages']):
+        # build a base Page instance from the exported content (so that we pick up its title and other
+        # core attributes)
+        page = Page.from_serializable_data(page_record['content'])
+        original_path = page.path
+        original_id = page.id
+
+        # clear id and treebeard-related fields so that they get reassigned when we save via add_child
+        page.id = None
+        page.path = None
+        page.depth = None
+        page.numchild = 0
+        page.url_path = None
+        page.content_type = page_content_type
+        parent_page.add_child(instance=page)
+
+        # Custom Code to add the new pk back into the original page record.
+        page_record['content']['pk'] = page.pk
+
+        pages_by_original_id[page.id] = page
+
+    for (i, page_record) in enumerate(import_data['pages']):
```

Comment:
Seems like this could be optimized to not have to loop through pages twice.

Reply:
For simple cases yes, but it would probably break on complex page imports if we didn't lay out the structure of base pages first.  That was wagtail's rationale for doing it this way at least.